src/libYARP_dev/src/devices/ControlBoardWrapper/RPCMessagesParser.cpp

```diff
@@ -211,7 +211,8 @@ void RPCMessagesParser::handleControlModeMsg(const yarp::os::Bottle& cmd,
                     {
                         modes[i] = modeList.get(i).asVocab();
                     }
-                    *ok = rpc_iCtrlMode2->setControlModes(n_joints, js, modes);
+                    if(rpc_iCtrlMode2)
+                        *ok = rpc_iCtrlMode2->setControlModes(n_joints, js, modes);
```

Comment:
If for any reason the `rpc_iCtrlMode2` pointer is null, than this code need either to fail abruptly in debug mode (i.e. `assert(rpc_iCtrlMode2)` ) if it is impossible for the logic of the code, or return an error, i.e. add an `else *ok = false`.

cc @barbalberto can probably comment on which solution he prefers.

Reply:
Yes, I'd add the `else /ok=false` case

Reply:
ok done

Reply:
Fixed in https://github.com/robotology/yarp/pull/1387/commits/6b79a0c894708c3ef19992997f1ad1ba8968f39e
Also added the else condition to other sub-cases of the `default` case